### DISCUSSION TITLE
Feature Requests: Different Priced/Type of Plan, Coupon Codes

### DISCUSSION THREAD

**Discussion by benf2004** [Glench/ExtPay#31] [Ideas]
Hi! Thanks so much for making ExtPay! I will plan on implementing it when my extension launches, but I have two (and a half) feature requests:

1. Allow differently priced plans for the same extension. So if I had a premium tier vs a basic tier, I could charge for both without creating a new extension. Also, I would love to offer both a yearly & monthly plan (or a monthly & one-time payment) so that I can incentivize users to go for the yearly/lifetime plan with a discount.
2. Much of what I suggested could be implemented with Stripe coupon codes. I would love to be able to offer discounts or ways to track affiliate earnings through ExtPay, which could be implemented through coupon codes. 

Thanks so much!

**Comment by surtic86** [comment]
Yes i would also love to have this Option to have more Option for like a Basic, Premium and Business Subscriptions.

**Comment by burkybang** [comment]
Would be very useful!

**Reply by benf2004** [reply]
@burkybang @surtic86 Hey everyone! I have figured out how you can add multiple plans to your extension (and confirmed with @Glench that "if it works it works"). In my case, I'm only offering two payment options (monthly & lifetime), so for me it's not too complicated. It's a bit of a jerry-rig, but it's not too bad.

1. Create multiple extensions on [extensionpay.com](url), one for each "plan" you want to offer.
2. Do not automatically call `extpay.startBackground()`, because you're not sure which plan the user has selected. Instead, once a user has selected a plan, send a message to the background script with whatever plan they chose and then startBackground(). After the background has started, you can call `extpay.openPaymentPage()`
3. Whenever a user wants to switch plans, first you **have to** make sure they have canceled the other plan. I made a big warning popup dialog that explains their other plan does not automatically cancel. 
4. Then, after the user has canceled & confirmed their cancellation, you have to remove the extension pay variables from sync storage:
`chrome.storage.sync.remove(['extensionpay_api_key','extensionpay_installed_at', "extensionpay_user"]`. **Warning: This will immediately sign out the user of their 'paid' features.** While they can sign back in, make sure they understand by clicking continue, they will be immediately signed out. Some people may click the "buy now" button and then change their mind before they complete the purchase. That can be a hassle if you don't ensure the user knows what is happening.
5. (EDIT) Make sure you once again pass a message to the background with the new plan choice so you can call `extpay.startBackground()`. 
6. I also opted to store an additional variable in Chrome storage called `[extension_name]_[plan_name]`. So I retrieve the plan from chrome storage and then set `const extpay = ExtPay(plan_name)` This way, in my content & background scripts I can get the appropriate plan to show the correct features/upgrade buttons. 

If you have any questions or want some more examples, don't be afraid to reach out!

**Reply by burkybang** [reply]
Very creative idea
